### DISCUSSION TITLE
Solid.js Support

### DISCUSSION THREAD

**Discussion by charkour** [payloadcms/payload#2298] [Feature Requests & Ideas]
Hey Team,

Thoughts on how difficult/hard it would be to allow for using custom Solid components in the Payload Admin UI?

Since payload is an API and UI, could we make an entirely separate UI built with Solid?

Thanks

**Comment by charkour** [comment]
To answer my question, it's possible to do this by running the Payload and building a UI layer a-top the existing API.

**Reply by matthijs166** [reply]
Yes! we do this all the time with Svelte.

Using the REST API gives you essentially the same capabilities that the Admin UI uses internally.

If you want some extra convenience, you can also use the Payload REST API SDK:
https://payloadcms.com/docs/rest-api/overview#payload-rest-api-sdk
